agent/acs/client/acs_client.go

```diff
@@ -53,7 +53,7 @@ func New(url string, region string, credentialProvider *credentials.Credentials,
 // call as unhandled requests will be discarded.
 func (cs *clientServer) Serve() error {
 	log.Debug("Starting websocket poll loop")
-	if cs.Conn == nil {
+	if !cs.IsConnected() {
 		return errors.New("nil connection")
```

Comment:
The error message should be changed to indicate that it's not connected instead of `nil connection`